test/test_logger.rb

```diff
@@ -27,15 +28,22 @@ def teardown
         if @logfile_io
             @logfile_io.close
         end
+        if @logfile
+            @logfile.close
+        end
     end
 
-    def logfile_path
-        @logfile_io ||= Tempfile.open('rock_logger_test.0.log')
+    def logfile_path(arg='/tmp/rock_logger_test.log')
+        @logfile_io ||= File.open(arg, "w+")
+        unless @logfile_io.path == arg
+          @logfile_io.close
+          @logfile_io = File.open(arg, "w+")
+        end
         @logfile_io.path
     end
 
-    def logfile
-        @logfile ||= Pocolog::Logfiles.open(logfile_path)
+    def logfile(file)
+        @logfile ||= Pocolog::Logfiles.open(logfile_path(file))
```

Comment:
This whole behind-the-scenes magic makes things **really** hard to follow and understand. Now that we can't assume that `.file` is a constant (or rather `.current_file` ;-)), change the test suite accordingly: open the log file from within the test suite at the point of use, without storing it at all as an attribute. This will remove the need to check for the `auto_*` attribute in all test cases, and make the subclass-and-set-auto-property-to-true a clean thing.

You're actually already mostly doing that when you do logfile(path).stream() everywhere.

Reply:
Agreed. I was quit unhappy with passing around arguments. Will strip this down to logfile(path).stream().